### DISCUSSION TITLE
Exploring the Possibility of Using Alpine as a Base Image

### DISCUSSION THREAD

**Discussion by DavoBR** [gotenberg/gotenberg#800] [Q&A] [answered]
Hello everyone,

I would like to open a discussion on the possibility of using alpine as a base image instead of debian:12-slim.

I have noticed that the debian:12-slim image has some known CVE vulnerabilities, and I am concerned about the security of the tool. Alpine is a smaller and more secure base image that could be a viable alternative.

To contribute to this discussion:

- I will share the results of a Cloud Guard analysis of the Debian:12-slim image. This analysis details the vulnerabilities present in the image and their potential impact on the security of the tool.

[Gotenberg8_CloudGuardResults.csv](https://github.com/gotenberg/gotenberg/files/14325724/Gotenberg8_CloudGuardResults.csv)

- I would like to know if anyone has tried using alpine or has any opinions on its viability as a base image.

- I would also like to hear the community's thoughts on the overall security of the tool and whether there are any other measures that could be taken to improve it.

I appreciate your comments and feedback on this topic.

**Comment by gulien** [comment]
Hello @DavoBR,

> I will share the results of a Cloud Guard analysis of the Debian:12-slim image. This analysis details the vulnerabilities present in the image and their potential impact on the security of the tool.

I've cleaned your upload, as I'm not a fan of XLSX files in the wild. Could you instead put a text variant directly in your post?

> I would like to know if anyone has tried using alpine or has any opinions on its viability as a base image.

Here's the list of current dependencies: https://github.com/gotenberg/gotenberg/blob/main/build/Dockerfile
IMO not viable because either:

1. Not all platforms are available.
2. Some dependencies do not exist.
3. Some dependencies are not up-to-date.

> I would also like to hear the community's thoughts on the overall security of the tool and whether there are any other measures that could be taken to improve it.

I feel like people are more and more complaining about this, mostly because of automation. They don't like their tool telling them "hey, I found these CVEs", so they come here saying "plz fix" but without an actual plan. Most of the time, those CVEs are on upstream dependencies, so even if I trigger a rebuild of the Gotenberg image, if there is no fix upstream, it's useless.

I could work on an alternative solution as described here https://github.com/gotenberg/gotenberg/pull/676, but I don't have the time currently to work on it.

**Reply by gulien** [reply]
Also, this is an open source project, everyone is free to schedule their own build.

**Comment by DavoBR** [comment, marked as answer]
> I've cleaned your upload, as I'm not a fan of XLSX files in the wild. Could you instead put a text variant directly in your post?

Sure, I have already edited the main post by adding a CSV version.

> I feel like people are more and more complaining about this, mostly because of automation. They don't like their tool telling them "hey, I found these CVEs", so they come here saying "plz fix" but without an actual plan. Most of the time, those CVEs are on upstream dependencies, so even if I trigger a rebuild of the Gotenberg image, if there is no fix upstream, it's useless.

I agree with you. My reason for starting this discussion is to find out if anyone else has attempted to address the CVEs or if there is a plan to remediate them. Most of the observations presented in the analysis refer to libraries in the Debian image. There are others that can be fixed by updating the components you use, such as Java, LibreOffice, Chromium, etc.

However, considering the possibility of using a distro with fewer vulnerabilities could be helpful. I have already started doing some tests using Alpine and the LibreOffice package. I am currently studying your unoconv scripts. I don't have much time to dedicate to this either, but we'll see how it goes and I'll share my results.

**Reply by slice-dweep** [reply]
Hey @DavoBR, did you find any solution?

**Reply by DavoBR** [reply]
> Hey @DavoBR, did you find any solution?

Hi, since the use of this software wasn’t approved internally, I had to explore alternatives and eventually ended up developing my own service to convert documents using LibreOffice.

My first version was built in Python, but I later migrated it to Java to simplify the implementation.

This version reflects known vulnerabilities reported in LibreOffice. 

Here’s the repository: https://github.com/DavoBR/document-converter
